app/controllers/file.rb

```diff
@@ -57,6 +57,10 @@ def slim(template, options={})
 
 				get "/?*", if: -> { resource.directory? || !resource.exist? } do
 					authorize resource, :index?
+
+					session["moves"] ||= {}
+					session["moves"][site.route] ||= []
```

Comment:
Couldn't this be `session["moves"] ||= { site.route => [] }`? If `session["moves"]` doesn't exist then it would be safe to safe to assign `site.route`.